### PR TITLE
solution 31/08

### PR DESCRIPTION
**Pull request by VadimKadasiev**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    ports:
+      - $SPRING_LOCAL_PORT:$SPRING_DOCKER_PORT
+      - $DEBUG_PORT:$DEBUG_PORT
+    environment:
+      SPRING_APPLICATION_JSON: '{
+        "spring.datasource.url": "jdbc:mysql://mysqldb:$MYSQLDB_DOCKER_PORT/$MYSQLDB_DATABASE",
+        "spring.datasource.username": "$MYSQLDB_USER",
+        "spring.datasource.password": "$MYSQLDB_ROOT_PASSWORD"
+      }'
+      JAVA_TOOL_OPTIONS: "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5005"
```
Comment (sarakhmen) [inline docker-compose.yml]: please use your DEBUG_PORT env var here

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
 spring.datasource.url=jdbc:mysql://localhost:3306/kadas?serverTimezone=UTC
 spring.datasource.username=root
-spring.datasource.password=1234
+spring.datasource.password=${DB_PASSWORD}
```
Comment (sarakhmen) [inline src/main/resources/application.properties]: by default spring cannot resolve such placeholders here from your .env file. Did you check it's working?

**Review by sarakhmen — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties (hunk 2 above)
Comment (VadimKadasiev) [reply src/main/resources/application.properties]: Checked it in different ways. Maybe there's something i misunderstand. Would appreciate more information of this issue.

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -1,6 +1,6 @@
 spring.datasource.url=jdbc:mysql://localhost:3306/kadas?serverTimezone=UTC
 spring.datasource.username=root
-spring.datasource.password=1234
+spring.datasource.password=${MYSQLDB_ROOT_PASSWORD}
```
Comment (okuzan) [inline src/main/resources/application.properties]: no need in this change, mentor is right, for .env vars to be resolved in properties 
spring.config.import=optional:file:.env[.properties]

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    ports:
+      - $SPRING_LOCAL_PORT:$SPRING_DOCKER_PORT
+      - $DEBUG_PORT:$DEBUG_PORT
+    environment:
+      SPRING_APPLICATION_JSON: '{
+        "spring.datasource.url": "jdbc:mysql://mysqldb:$MYSQLDB_DOCKER_PORT/$MYSQLDB_DATABASE",
+        "spring.datasource.username": "$MYSQLDB_USER",
+        "spring.datasource.password": "$MYSQLDB_ROOT_PASSWORD"
+      }'
+      JAVA_TOOL_OPTIONS: "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:$DEBUG_PORT"
```
Comment (okuzan) [inline docker-compose.yml]: image
please always add empty line where you see this symbol

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+# Final stage
+FROM openjdk:22-jdk-slim
+WORKDIR application
+COPY --from=builder application/dependencies/ ./
+COPY --from=builder application/spring-boot-loader/ ./
+COPY --from=builder application/snapshot-dependencies/ ./
+COPY --from=builder application/application/ ./
+ENTRYPOINT ["java", "org.springframework.boot.loader.launch.JarLauncher"]
+EXPOSE 8080
```
Comment (okuzan) [inline Dockerfile]:
```suggestion
EXPOSE 8080

```

**Review by okuzan — CHANGES_REQUESTED** [review]
remove .env from repo, it contains secrets  - git rm --cached .env
add .env.template with actual values removed
add .env to .gitignore

**Review by Elena-Bruyako — APPROVED** [review]
well done

**Pull request merged**